atlasdb-ete-tests/src/main/java/com/palantir/atlasdb/lock/SimpleEteLockWatchResource.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    @Override
+    public Optional<CommitUpdate> endTransaction(TransactionId transactionId) {
+        TransactionAndCondition txnAndCondition = activeTransactions.remove(transactionId);
+        txnAndCondition.transaction().commit();
+        txnAndCondition.transaction().finish(unused -> null);
+        txnAndCondition.condition().cleanup();
+        return Optional.ofNullable(txnAndCondition.condition().getCommitUpdate());
+    }
+
+    @Override
+    public void write(WriteRequest writeRequest) {
+        Transaction transaction = activeTransactions.get(writeRequest.id()).transaction();
+        Map<Cell, byte[]> byteValues = getValueMap(writeRequest.rows());
+        transaction.put(lockWatchTable, byteValues);
+    }
+
+    @Override
+    public TransactionsLockWatchUpdate getUpdate(GetLockWatchUpdateRequest updateRequest) {
+        return lockWatchManager.getUpdateForTransactions(updateRequest.startTimestamps(), updateRequest.version());
+    }
+
+    @Override
+    public void setTable(String tableName) {
+        table = tableName.replace("\"", "");
+        lockWatchTable = TableReference.create(NAMESPACE, table);
+        createTable();
+    }
+
+    private void createTable() {
+        KeyValueService keyValueService = transactionManager.getKeyValueService();
+        keyValueService.createTable(lockWatchTable, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        transactionManager
+                .getLockWatchManager()
+                .registerWatches(ImmutableSet.of(LockWatchReferences.entireTable(lockWatchTable.getQualifiedName())));
+    }
+
+    private Map<Cell, byte[]> getValueMap(Set<String> rows) {
+        return KeyedStream.of(rows)
+                .mapKeys(PtBytes::toBytes)
+                .mapKeys(row -> Cell.create(row, COLUMN))
+                .map(unused -> VALUE)
+                .collectToMap();
+    }
+
+    @Value.Immutable
+    interface TransactionAndCondition {
+        @Value.Parameter
+        OpenTransaction transaction();
+
+        @Value.Parameter
+        CommitUpdateCondition condition();
+
+        static TransactionAndCondition of(OpenTransaction transaction, CommitUpdateCondition condition) {
+            return ImmutableTransactionAndCondition.of(transaction, condition);
+        }
+    }
+
+    class CommitUpdateCondition implements PreCommitCondition {
+        private final AtomicReference<CommitUpdate> commitUpdate = new AtomicReference<>();
+        private Optional<Long> startTs = Optional.empty();
+
+        CommitUpdateCondition() {}
+
+        void setStartTs(long startTs) {
+            this.startTs = Optional.of(startTs);
+        }
+
+        CommitUpdate getCommitUpdate() {
+            return commitUpdate.get();
+        }
+
+        @Override
+        public void throwIfConditionInvalid(long timestamp) {
+            if (startTs.filter(ts -> ts != timestamp).isPresent()) {
+                commitUpdate.set(lockWatchManager.getCommitUpdate(startTs.get()));
```

Comment:
This is necessary as we only want the commit update when we are passed the commit timestamp, not the start.